include/device.h

```diff
@@ -737,8 +737,8 @@ BUILD_ASSERT(sizeof(device_handle_t) == 2, "fix the linker scripts");
 		.pm = {						        \
 			.flags = ATOMIC_INIT(COND_CODE_1(		\
 					DT_NODE_EXISTS(node_id),	\
-					(DT_PROP(			\
-					node_id, wakeup_source)),	\
+					(DT_PROP_OR(			\
```

Comment:
@mbolivar-nordic I remember you suggesting `DT_PROP` here, can you check?

Reply:
Yep, initially it was just using  `DT_PROP_OR` but @mbolivar-nordic had a point during the review:

```
A couple of other things I noticed while looking this over.

Public service announcement regarding boolean properties from the DT_PROP documentation:

 * For properties whose bindings have the following types, this macro
 * expands to:
 * [...]
 * - boolean: 0 if the property is false, or 1 if it is true

There's no need to use DT_NODE_HAS_PROP or DT_PROP_OR with boolean properties. I see this happen a lot and it's not necessary. The type system knows what to do with these; using DT_PROP is safe.
```

I had to add then the DT_NODE_EXISTS because this is used for devices that are defined in the DT as well.

Reply:
it seems that in the generated header `child-binding`s don't define properties inherited from the parent.

Reply:
@mbolivar-nordic  if what I said is right, don't we have a problem ?

Reply:
the `child-binding` is one thing, but there is also `gpio-leds` so the assumption that all device nodes inherit from base.yaml shouldn't be assumed and thus I think its better to use DT_PROP_OR here.

Reply:
agree

Reply:
Just let see what @mbolivar-nordic  has to say since he suggested to use `DT_PROP` instead of using `DT_PROP_OR`. Btw, if it is going to use `DT_PROP_OR` this can be simplified as the original version:
```
			.flags = ATOMIC_INIT(DT_PROP_OR(                  \
					node_id, wakeup_source, 0)	<<           \
				        PM_DEVICE_FLAGS_WS_CAPABLE),  \
```

Reply:
Hmm, @mbolivar point in that comment is that `DT_PROP_OR` should behavior just like `DT_PROP` with boolean properties, so we may be missing something.

Reply:
I think that his comment about `DT_PROP` and `DT_PROP_OR` being the same for boolean is with the assumption that they property is defined for the node in the binding.

Reply:
> the assumption that they property is defined for the node in the binding

OK, yes, that's right. If it is possible that there are nodes without this property defined in the binding, then we can't benefit from DT_PROP as defined in the binding. I did not realize that there were devices without this property in their nodes' bindings. The original pushback from @ceolin was that some callers were passing DT_INVALID_NODE here, not that there were valid nodes without `wakeup-source` properties.

@galak can you add a comment here explaining why DT_PROP_OR() is being used?

I'm pushing for DT_PROP() on boolean properties so we get build errors if there are typos or other issues with using the wrong binding etc. This seems like a valid exception but it's worth documenting in a comment IMO.

Reply:
@mbolivar-nordic is that possible if it is included by base.yaml? https://github.com/zephyrproject-rtos/zephyr/blob/f3435b75efbfd7614ff873f8bfa90a031d6d27e8/dts/bindings/base/base.yaml#L3

Reply:
The problem AFAIU is that it child bindings don't have the parent properties defined. I don't know if that is a real problem, but it makes at least hard to work with child-bindings because it is necessary to access the parent to know if it has the property. Would be much easier if child-bindings inherit (define the macros in the generated file) properties from the parent.

Reply:
> > the assumption that they property is defined for the node in the binding
> 
> OK, yes, that's right. If it is possible that there are nodes without this property defined in the binding, then we can't benefit from DT_PROP + boolean property as defined in the binding. I did not realize that there were devices without this property in their nodes' bindings. The original pushback from @ceolin was that some callers were passing DT_INVALID_NODE here, not that there were valid nodes without `wakeup-source` properties.
> 
> @galak can you add a comment here explaining why DT_PROP_OR() is being used?
> 
> I'm pushing for DT_PROP() on boolean properties so we get build errors if there are typos or other issues with using the wrong binding etc. This seems like a valid exception but it's worth documenting in a comment IMO.

Indeed, get build errors is indeed a good thing. I pushed back about `DT_INVALID_NODE` because I thought all properties in `base.yaml` was available in all devices.

Reply:
Added a comment about use of DT_PROP_OR()